File: samples/snippets/csharp/VS_Snippets_CLR_System/system.Random.Ctor/CS/ctor4.cs
// <Snippet4>
using System;
using System.Threading;

public class Example
{
   public static void Main()
   {
      Random rand1 = new Random((int) DateTime.Now.Ticks & 0x0000FFFF);
      Random rand2 = new Random((int) DateTime.Now.Ticks & 0x0000FFFF);
      Thread.Sleep(20);
      Random rand3 = new Random((int) DateTime.Now.Ticks & 0x0000FFFF);
      ShowRandomNumbers(rand1);
      ShowRandomNumbers(rand2);
      ShowRandomNumbers(rand3);
   }

   private static void ShowRandomNumbers(Random rand)
   {
      Console.WriteLine();
      byte[] values = new byte[4];
      rand.NextBytes(values);
      foreach (var value in values)
         Console.Write("{0, 5}", value);

      Console.WriteLine(); 
   }
}
// The example displays output similar to the following:
//   145  214  177  134  173
// 
//   145  214  177  134  173
// 
//   126  185  175  249  157
// </Snippet4>

File: samples/snippets/visualbasic/VS_Snippets_CLR_System/system.Random.Ctor/VB/ctor4.vb
' Visual Basic .NET Document
Option Strict On

' <Snippet4>
Imports System.Threading

Module RandomNumbers
   Public Sub Main()
      Dim rand1 As New Random(CInt(Date.Now.Ticks And &h0000FFFF))
      Dim rand2 As New Random(CInt(Date.Now.Ticks And &h0000FFFF))
      Thread.Sleep(20)
      Dim rand3 As New Random(CInt(Date.Now.Ticks And &h0000FFFF))
      ShowRandomNumbers(rand1)
      ShowRandomNumbers(rand2)
      ShowRandomNumbers(rand3)
   End Sub
   
   Private Sub ShowRandomNumbers(rand As Random)
      Console.WriteLine()
      Dim values(4) As Byte
      rand.NextBytes(values)
      For Each value As Byte In values
         Console.Write("{0, 5}", value)
      Next      
      Console.WriteLine() 
   End Sub
End Module
' The example displays output similar to the following:
'      145  214  177  134  173
'    
'      145  214  177  134  173
'    
'      126  185  175  249  157
' </Snippet4>

